server/templates/importer.go

```diff
@@ -9,9 +9,15 @@ func DiscoverTemplateOrder(templateBasePath string) ([][]string, error) {
 	dependenciesGraph := NewDependencyGraph()
 
 	templateDirEntries, err := os.ReadDir(templateBasePath)
+
+	if len(templateDirEntries) == 0 {
```

Comment:
Not following the need for this line, if there are no template entries, the `for` loop below will do nothing and the function will return the same thing, or am I missing something here?

Reply:
1. The code will actually return a non-nil err if this isn't checked, which will cause the calling code in `starter.go - 38, 40: templateOrderArray, err := templates.DiscoverTemplateOrder(templatePath)` to execute a `Fatal`, triggering an exit of the server process.

2. This resolves having no template directories for the time being, while expecting this code to be replaced in the future if the 3-segmented JSON files are no longer used in favor of a "mainTemplate" that's submitted to the _"Azure Deployment Manager"_ we're going to consume.

Example of no longer being necessary in the future:
```golang
25      filepath.Join(templateBasePath, name, name+".dependencies.json")
```

Reply:
That's a good catch, however, if we only check the length of the directory entries, then we are ignoring an error, which could be legitimate use case or unexpected error, and then we need to know about it.

Reply:
Agreed. So what error would require the server to go into a fatal state? In this case, if there isn't any templates it just shuts down.

Reply:
@tznamena Having the server fatally shutdown causes an unstable state for the web UI / frontend. If there are no templates, I can instead update this PR to cause the server to get a "server status" endpoint similar to the current `/status` endpoint, which will describe the overall state of the server. This way, the customer can clearly identify a failure along with an automatic shutdown time in the UI (graceful), and SRE will be notified. What do you think?

Reply:
That's an awesome idea, makes the server always come up and it can report its state. Along this line of thinking, we can add same logic around the code that's connecting to azure (but that would be for separate PR)